### PR TITLE
DHFPROD-7905: Persisting graph config data using new endpoint

### PR DESCRIPTION
**Pull request by xnikhil08**
#### Checklist: 
```diff
- Note: do not change the below
```

-  ##### Owner:

- [x] JIRA_ID included in all the commit messages
- [x] PR title is in the format JIRA_ID:Title
- [x] Rebase the branch with upstream
- [x] Squashed all commits into a single commit
- [x] Code passes ESLint tests
- [x] Added Tests
- [x] Run UI tests
  

- ##### Reviewer:

- [x] Reviewed Tests

### CODE REVIEW

**Review by wooldridge — CHANGES_REQUESTED** [review]
When I edit the node color with the side panel open, I don't see an immediate change in the node color. It then changes when I close the side panel:

https://watch.screencastify.com/v/lJFF3pfE7g3rRkHY3SVP

Observations while creating and then deleting an entity:

When I create a new entity, there is a draft version of that entity created but I can move away from the modeling with without a warning. When I return to the modeling view there is a warning. (Also, should entity creation require a publish step?)

When I delete an entity, I get what appears to be a modeling update. (A spinner on the button.) But the result is a new draft entity in the database for that deleted entity. Then I have to Publish to finish deleting. Is that expected? Shouldn't a deletion just delete everything for that entity and not require and additional publish step?

Screencast: https://watch.screencastify.com/v/Tmh4tbGt0RBnhIXQHJmO

File: marklogic-data-hub-central/ui/src/api/modeling.ts [hunk 1]
```diff
@@ -43,3 +34,11 @@ export const updateEntityModels = async (entityModifiedArray: EntityModified[])
 export const publishDraftModels = async () => {
   return await axios.put(`/api/models/publishDraftModels`);
 };
+
+export const hubCentralConfig = async () => {
```
[the remark this answers is not in the document]
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/api/modeling.ts]: Done.

File: marklogic-data-hub-central/ui/src/components/modeling/entity-type-modal/entity-type-modal.tsx [hunk 2]
```diff
@@ -142,10 +163,15 @@ const EntityTypeModal: React.FC<Props> = (props) => {
       } else {
         handleError(error);
       }
-      toggleLoading(false);
     }
   };
 
+  const UpdateHubCentralConfig = async (entityName, color) => {
```
[the remark this answers is not in the document]
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/modeling/entity-type-modal/entity-type-modal.tsx]: Done.

File: marklogic-data-hub-central/ui/src/components/modeling/entity-type-table/entity-type-table.test.tsx [hunk 3]
```diff
@@ -66,6 +68,7 @@ describe("EntityTypeModal Component", () => {
           editEntityTypeDescription={jest.fn()}
           updateEntities={jest.fn()}
           updateSavedEntity={jest.fn()}
+          hubCentralConfig={hubCentralConfig}
```
[the remark this answers is not in the document]
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/modeling/entity-type-table/entity-type-table.test.tsx:71]: Here **hubCentralConfig** is an object containing the configuration, not the function to get the object.

File: marklogic-data-hub-central/ui/src/components/modeling/entity-type-table/entity-type-table.test.tsx [hunk 4]
```diff
@@ -151,6 +154,7 @@ describe("EntityTypeModal Component", () => {
           editEntityTypeDescription={editMock}
           updateEntities={jest.fn()}
           updateSavedEntity={jest.fn()}
+          hubCentralConfig={hubCentralConfig}
```
[the remark this answers is not in the document]
Comment (xnikhil08) [reply marklogic-data-hub-central/ui/src/components/modeling/entity-type-table/entity-type-table.test.tsx:157]: Same as previous comment. **hubCentralConfig** is not a function here.

**Comment by xnikhil08** [comment]
> When I edit the node color with the side panel open, I don't see an immediate change in the node color. It then changes when I close the side panel:
> 
> https://watch.screencastify.com/v/lJFF3pfE7g3rRkHY3SVP
> 
> Observations while creating and then deleting an entity:
> 
> When I create a new entity, there is a draft version of that entity created but I can move away from the modeling with without a warning. When I return to the modeling view there is a warning. (Also, should entity creation require a publish step?)
> 
> When I delete an entity, I get what appears to be a modeling update. (A spinner on the button.) But the result is a new draft entity in the database for that deleted entity. Then I have to Publish to finish deleting. Is that expected? Shouldn't a deletion just delete everything for that entity and not require and additional publish step?
> 
> Screencast: https://watch.screencastify.com/v/Tmh4tbGt0RBnhIXQHJmO

@wooldridge - Fixed the issue where Alert was not visible on new entity creation. Also, regarding queries related to the creation and deletion of entities; It is designed this way to avoid having any modifications in the model, reach other tiles in HC. We can revisit this behavior if needed later or perhaps based on user feedback.

**Review by briantang — DISMISSED** [review]
Functionality looks good to me with the relocation of graph config. Color and node positioning changes no longer trigger the alert but are persisted.

**Review by briantang — DISMISSED** [review]

**Review by wooldridge — CHANGES_REQUESTED** [review]
I'm seeing config values for an entity accumulate in the config file even when the associated entity has been deleted. As a result, if I create an entity, delete it, and then create it again, the second new version gets properties of the first version (e.g., color): 
https://watch.screencastify.com/v/sAIjUolUHC5opDaapbBM

I would think removing entity objects from the config file upon entity deletion would fix this. But I'm not sure if this is supported by the endpoint! Can we ignore this edge case for now? Or can we remove the entity object from the config object in the UI code and resave?

**Comment by xnikhil08** [comment]
> https://watch.screencastify.com/v/sAIjUolUHC5opDaapbBM

@wooldridge - This seems like a reasonable request. I hav e created a separate ticket to handle deletion of hubCentral config for an entity type while deleting the entity type (https://project.marklogic.com/jira/browse/DHFPROD-7956).
@ryanjdew - Just FYI

**Review by briantang — DISMISSED** [review]

**Review by wooldridge — CHANGES_REQUESTED** [review]
Seeing an error in the UI display upon trying to create a new entity:

https://watch.screencastify.com/v/CnTWli42BHiZJVWY0PsL

Complaining about draft-entity-model-validate-trigger.xqy. :-(

**Comment by xnikhil08** [comment]
> draft-entity-model-validate-trigger.xqy

@wooldridge - Looks like the file draft-entity-model-validate-trigger.xqy is missing in your database. Try redeploying the project after publishing the branch changes to local maven..

**Review by wooldridge — APPROVED** [review]

**Review by briantang — APPROVED** [review]

**Review by briantang — APPROVED** [review]

**Review by AkshayBajajML — APPROVED** [review]

**Pull request merged**